aws/data_source_aws_mq_broker.go

```diff
@@ -250,42 +251,113 @@ func dataSourceAwsMqBroker() *schema.Resource {
 }
 
 func dataSourceAwsmQBrokerRead(d *schema.ResourceData, meta interface{}) error {
-	if brokerId, ok := d.GetOk("broker_id"); ok {
-		d.SetId(brokerId.(string))
-	} else {
-		conn := meta.(*AWSClient).mqconn
-		brokerName := d.Get("broker_name").(string)
+	conn := meta.(*AWSClient).mqconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
 
-		input := &mq.ListBrokersInput{}
+	input := &mq.ListBrokersInput{}
 
-		err := conn.ListBrokersPages(input, func(page *mq.ListBrokersResponse, lastPage bool) bool {
-			if page == nil {
-				return !lastPage
-			}
+	var results []*mq.BrokerSummary
 
-			for _, brokerSummary := range page.BrokerSummaries {
-				if brokerSummary == nil {
-					continue
-				}
+	err := conn.ListBrokersPages(input, func(page *mq.ListBrokersResponse, lastPage bool) bool {
+		if page == nil {
+			return !lastPage
+		}
 
-				if aws.StringValue(brokerSummary.BrokerName) == brokerName {
-					d.Set("broker_id", brokerSummary.BrokerId)
-					d.SetId(aws.StringValue(brokerSummary.BrokerId))
-					return false
-				}
+		for _, brokerSummary := range page.BrokerSummaries {
+			if brokerSummary == nil {
+				continue
 			}
 
-			return !lastPage
-		})
+			if v, ok := d.GetOk("broker_id"); ok && v.(string) != aws.StringValue(brokerSummary.BrokerId) {
+				continue
+			}
 
-		if err != nil {
-			return fmt.Errorf("error listing MQ Brokers: %w", err)
-		}
+			if v, ok := d.GetOk("broker_name"); ok && v.(string) != aws.StringValue(brokerSummary.BrokerName) {
+				continue
+			}
 
-		if d.Id() == "" {
-			return fmt.Errorf("Failed to determine mq broker: %s", brokerName)
+			results = append(results, brokerSummary)
 		}
+
+		return !lastPage
+	})
+
+	if err != nil {
+		return fmt.Errorf("error listing MQ Brokers: %w", err)
+	}
+
+	if len(results) != 1 {
+		return fmt.Errorf("Search returned %d results, please revise so only one is returned", len(results))
+	}
+
+	brokerId := aws.StringValue(results[0].BrokerId)
+
+	output, err := conn.DescribeBroker(&mq.DescribeBrokerInput{
+		BrokerId: aws.String(brokerId),
+	})
+
+	if err != nil {
+		return fmt.Errorf("error reading MQ broker (%s): %w", brokerId, err)
+	}
+
+	if output == nil {
+		return fmt.Errorf("empty response while reading MQ broker (%s)", brokerId)
+	}
+
+	d.SetId(brokerId)
+
+	d.Set("arn", output.BrokerArn)
+	d.Set("authentication_strategy", output.AuthenticationStrategy)
+	d.Set("auto_minor_version_upgrade", output.AutoMinorVersionUpgrade)
+	d.Set("broker_name", output.BrokerName)
```

Comment:
I believe we still need to set `broker_id` attribute:

```
=== CONT  TestAccDataSourceAWSMqBroker_basic
    data_source_aws_mq_broker_test.go:21: Step 2/2 error: Check failed: 1 error occurred:
                * Check 1/2 error: data.aws_mq_broker.by_name: Attribute "broker_id" not set, but "id" is set in aws_mq_broker.acctest as "b-230d2c00-d06b-476a-8e27-b273217fd494"
```

```suggestion
	d.Set("broker_id", brokerId)
	d.Set("broker_name", output.BrokerName)
```

Reply:
ohhh nice catch!